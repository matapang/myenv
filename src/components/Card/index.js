import React, {Component} from 'react';
import {Card as BCard} from '@blueprintjs/core';


class CardNew extends Component {
    render() {
        const {children, className,...rest} = this.props;
        const allClassName = `bg-main pt-elevation-2 ${className || ''}`;
        return (
            <BCard className={allClassName} {...rest}>
                {children}
            </BCard>
        );
    }
}

export default CardNew;