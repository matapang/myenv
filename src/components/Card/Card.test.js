import React from 'react';

import renderer from 'react-test-renderer';
import Card from './index';


it('Card should match snapshot', () => {  
    const component = renderer.create(
      <Card>I am a child of this Card</Card>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
});


it('Card append passed class name with class name of bg-main pt-elevation-2 ', () => {  
  const component = renderer.create(
    <Card className="d-flex" />,
  );
  let tree = component.toJSON();  
  expect(tree.props.className).toEqual("pt-card pt-elevation-0 bg-main pt-elevation-2 d-flex")
});