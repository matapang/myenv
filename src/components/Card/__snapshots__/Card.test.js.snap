// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="pt-card pt-elevation-0 bg-main pt-elevation-2 "
>
  I am a child of this Card
</div>
`;
